File: frontend/src/algorithms/regime/strategies/safety/stale-data.ts
import type { RegimeMarketContext, RegimeRawStrategySignal } from "../../types.ts";
import { safetyGate } from "../base.ts";

export function staleDataGate(market: RegimeMarketContext): RegimeRawStrategySignal {
  if (market.contextFeeds.quoteFreshness.status === "stale") {
    return safetyGate(false, "Stale quote freshness blocks new Regime entries");
  }
  if (market.contextFeeds.quoteFreshness.status === "fresh") {
    return safetyGate(true, "Quote freshness is within Regime limits");
  }
  return safetyGate(true, "Stale-data check has no quote freshness feed; no stale condition detected in supplied snapshot");
}

File: backend/app/algorithms/regime/strategies/safety/stale_data.py
def evaluate(snapshot, classification):
    status = snapshot.context_feeds["quoteFreshness"].get("status")
    return "Hold", 1.0 if status == "stale" else 0.5, "regime.safety.stale_data" if status == "stale" else "regime.safety.clear", {"quoteFreshness": status}

